core/external_search.py

```
@@ -1913,6 +1918,138 @@ def _hypothesize(cls, hypotheses, query, boost, filters=None, **kwargs):
+        return self.FIELD_MAPPING[name].get("keyword") == True
+
+    def _nested_path(self, name: str) -> Union[str, None]:
+        return self.FIELD_MAPPING[name].get("path")
+
+    def _parse_json_query(self, query: Dict):
+        """Eventually recursive json query parser"""
+        es_query = None
+
+        leaves = {self.QueryLeaf.KEY, self.QueryLeaf.VALUE}
+        if set(query.keys()).intersection(leaves) == leaves:
+            es_query = self._parse_json_leaf(query)
+        elif {self.Conjunctives.AND, self.Conjunctives.OR}.issuperset(query.keys()):
+            es_query = self._parse_json_join(query)
+
+        return es_query
+
+    def _parse_json_leaf(self, query):
+        op = query.get(self.QueryLeaf.OP, self.Operators.EQ)
+
+        key = query[self.QueryLeaf.KEY]
+        nested_path = self._nested_path(key)
+        if self._is_keyword(key):
+            key = query[self.QueryLeaf.KEY] + ".keyword"
+
+        value = query[self.QueryLeaf.VALUE]
+
+        es_query = None
+
+        if op is self.Operators.EQ:
+            es_query = Term(**{key: value})
+        elif op is self.Operators.NEQ:
+            es_query = Bool(must_not=[Term(**{key: value})])
+        elif op in {
+            self.Operators.GT,
+            self.Operators.GTE,
+            self.Operators.LT,
+            self.Operators.LTE,
+        }:
+            es_query = Range(**{key: {op: value}})
+
+        # For nested paths
+        if nested_path:
+            es_query = Nested(path=nested_path, query=es_query)
+
+        return es_query
+
+    def _parse_json_join(self, query):
+        join = list(query.keys())[0]
+        to_join = []
+        for query_part in query[join]:
+            q = self._parse_json_query(query_part)
+            to_join.append(q)
+
+        if join == self.Conjunctives.AND:
+            joined_query = Bool(must=to_join)
+        elif join == self.Conjunctives.OR:
+            joined_query = Bool(should=to_join)
```

Same goes here. We may never add a new `Conjunctive`, but the code should fail here if we ever do rather than being silently wrong.